src/dask_awkward/lib/core.py

```diff
@@ -959,15 +960,33 @@ def __getitem__(self, where: Any) -> AwkwardDaskCollection:
 
     def _call_behavior_method(self, method_name: str, *args: Any, **kwargs: Any) -> Any:
         """Call a behavior method for an awkward array.
-        Please note that the raw dask_awkward arguments are forwarded to the
-        awkward Mixin class, and the user must deal with when to make calls
-        to map_partitions.
+        If the function signature has __dunder__ parameters it is assumed that the
+        user wants to do the map_partitions dispatch themselves and the _meta's
+        behavior is called.
+        If there are no __dunder__ parameters in the function call then the function
+        is wrapped in map_partitions automatically.
         """
         if hasattr(self._meta, method_name):
-            array_context = kwargs.pop("__dask_array__", self)
-            return getattr(self._meta, method_name)(
-                *args, __dask_array__=array_context, **kwargs
+            themethod = getattr(self._meta, method_name)
+            thesig = inspect.signature(themethod)
+            ndunder = sum(
+                [
+                    1
+                    if len(param) > 4
+                    and param.startswith("__")
+                    and param.endswith("__")
+                    else 0
+                    for param in thesig.parameters
+                ]
+            )
+            if ndunder > 0:
+                return themethod(*args, **kwargs)
+            return self.map_partitions(
+                _BehaviorMethodFn(method_name, **kwargs),
+                *args,
+                label=hyphenize(method_name),
```

Comment:
@lgray this looks like the same problem (3) as I was describing, albeit more general in that you're not just dealing with specifying the associative operation.

I still don't understand your use case, but I've not spent a great deal of time yet to figure it out.

I'm worried that this is solving too specific a problem; I'd like to find the more general problem and solve that. I will give this more thought, but I'm on holiday/thesis submission the next week, so time is thin on the gruond:/